### DISCUSSION TITLE
G6 Problems on Oneplus/Oppo Phones - Solved

### DISCUSSION THREAD

**Discussion by Miquest** [NightscoutFoundation/xDrip#2400] [Show and tell]
Hi,
I had some problems with my Dexcom G6 and xDrip since the OxygenOS/ColorOS 12 Update.
My Phone recived only a reading from my G6 every 10 Minutes, and that was really annoying. Until today!

You need this app, the Sensor Code:

https://docs.google.com/forms/d/e/1FAIpQLScD76G0Y-BlL4tZljaFkjlwuqhT83QlFM5v6ZEfO7gCU98iJQ/viewform?fbzx=2196386787609383750&fbclid=IwAR2aL8Cps1s6W8apUVK-gOqgGpA-McMPJj9Y8emf_P0-_gAsmJs6QwAY-o0

And the latest xDrip Nightly build, to use the Companion Bluetooth mode.

First, start the new Sensor in the Dexcom BYODA App, if you've already started one, the Dexcom App will recognize that.
You'll have to type the Sensor Code and the Transmitter ID again.

Then set the data source to Companion App and activate the Companion Bluetooth option.
Wait for the next reading, the BG readings should be in your xDrip Plus application every 5 minutes.
If you have questions, you're welcome to ask me them in this discussion!

Kind regards,
Thorin

**Comment by Navid200** [comment]
Wouldn't it be more helpful if you report the problem here and let us fix it?
After all, this is the xDrip repository.

You are posting in the xDrip repository telling people not to use it because ...?!

**Reply by Miquest** [reply]
I tell people to use it in combination with the original Dexcom App to solve a problem.

I reported the bug several times for months, but nothing happened.
You always told me to check my settings. Nothing really helpful.


So I had to look for other methods to solve my problem, and I want to share my method with other's.
I didn't want to uninstall xDrip Plus, because I like it so much. That's a simple way to use it with 5minute reading on OxygenOS/ColorOS 12.

**Reply by Miquest** [reply]
That's a workaround and no way telling people to use another application INSTEAD of xDrip Plus.

**Reply by Navid200** [reply]
If you have reported the problem in these discussions, please add a link to it in your first post.
If you have never reported the problem in these discussions, please do.

What I tell people in the facebook group does not represent the official xDrip stance.  It cannot.  I am one person.  I do my best to help.  But, there is a lot I don't know anything about.  I often tell people that some (main) xDrip developers don't even have a facebook account.  And I ask people to report unresolved problems in the xDrip discussions (here) for that reason.

Thanks for posting here.  
Even if you are happy with what you have found, please tell us about the original problem.

**Reply by Navid200** [reply]
If you decide to use the byoda, there is another way for passing data to xDrip:
https://navid200.github.io/xDrip/docs/DexcomAppxDrip.html

I am curious which one backfills.
You can enable a new setting that may enable backfill for companion:
Settings &#8722;> Less common settings &#8722;> Bluetooth Settings &#8722;> Companion Bluetooth
This is new (October 15, 2022) and is still being tested.

**Reply by Miquest** [reply]
https://github.com/NightscoutFoundation/xDrip/discussions/2106

**Reply by Miquest** [reply]
That's my description of the original problem...

**Reply by Miquest** [reply]
> If you decide to use the byoda, there is another way for passing data to xDrip: https://navid200.github.io/xDrip/docs/DexcomAppxDrip.html
> 
> I am curious which one backfills. You can enable a new setting that may enable backfill for companion: Settings −> Less common settings −> Bluetooth Settings −> Companion Bluetooth This is new (October 15, 2022) and is still being tested.

Maybe I have another version of the Dexcom BYODA.
